app/stac_api/models.py

```diff
@@ -229,6 +231,52 @@ def update_extent(self, item_properties_datetime):
         #     raise ValidationError(_("Updating the collection extent due to item update failed."))
         pass  # pylint: disable=unnecessary-pass
 
+    def update_bbox_extent(self, action, item_geom, item_id):
+        '''
+        updates the collection's spatial extent when an item is updated.
+        :param action: eighter up (insert, update) or rm (delete)
+        :param geometry: the geometry of the item
+        :param item_id the id of the item being treated
+        This function generates a new extent regarding all the items with the same
+        collection foreign key. If there is no spatial bbox yet, the one of the geometry of the
+        item is being used.
+        '''
+
+        # insert (as item_id is None)
+        if action == 'up' and item_id is None:
+            # the first item of this collection
+            if self.extent_geometry is None:
+                self.extent_geometry = Polygon.from_bbox(GEOSGeometry(item_geom).extent)
+            # there is already a geometry in the collection a union of the geometries
+            else:
+                self.extent_geometry = Polygon.from_bbox(
+                    GEOSGeometry(self.extent_geometry).union(GEOSGeometry(item_geom)).extent)
+
+        # update
+        if action == 'up' and item_id:
+            # is the new bbox larger than (and covering) the existing
+            if Polygon.from_bbox(GEOSGeometry(item_geom).extent).covers(self.extent_geometry):
+                self.extent_geometry = Polygon.from_bbox(GEOSGeometry(item_geom).extent)
+            # we need to iterat trough the items
+            else:
+                qs = Item.objects.filter(collection_id=self.pk).exclude(id=item_id)
+                union_geometry = GEOSGeometry(item_geom)
+                for item in qs:
+                    union_geometry = union_geometry.union(item.geometry)
+                self.extent_geometry = Polygon.from_bbox(union_geometry.extent)
+
```

Comment:
I think this case is not necessary (or do I miss sth?): if the current 'cached' extent is bigger than the extent of the new item, then we don't need to do anything I think.

Reply:
when the updated extent is smaller than the current collection extent, it could be, that the collection extent has to become smaller - we don't know -> so we have to iterate

Reply:
Yeah true, didnt think about that one